File: examples/customer-address-jpa/src/main/java/io/zenwave360/example/core/inbound/CustomerService.java
package io.zenwave360.example.core.inbound;

import io.zenwave360.example.core.domain.Customer;
import io.zenwave360.example.core.inbound.dtos.CustomerSearchCriteria;
import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/** Inbound Service Port for managing [Customer]. */
public interface CustomerService {

    /** With Events: [CustomerEvent]. */
    public Customer createCustomer(Customer input);

    /** */
    public Optional<Customer> getCustomer(Long id);

    /** With Events: [CustomerEvent]. */
    public Optional<Customer> updateCustomer(Long id, Customer input);

    /** With Events: [CustomerEvent]. */
    public void deleteCustomer(Long id);

    /** */
    public Page<Customer> searchCustomers(CustomerSearchCriteria input, Pageable pageable);
}

File: examples/kustomer-address-jpa/src/main/kotlin/io/zenwave360/examples/kotlin/service/CustomerService.kt
package io.zenwave360.examples.kotlin.service

import io.zenwave360.examples.kotlin.domain.*
import io.zenwave360.examples.kotlin.service.dtos.*
import java.math.*
import java.time.*
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.Optional

/**
 * Inbound Service Port for managing [Customer].
 */
interface CustomerService {


         /**
      * 
      *
      * With Events: [CustomerEvent].

      */

     fun createCustomer(input: Customer): Customer


         /**
      * 
      *
      */

     fun getCustomer(id: Long): Optional<Customer>


         /**
      * 
      *
      * With Events: [CustomerEvent].

      */

     fun updateCustomer(id: Long, input: Customer): Optional<Customer>


         /**
      * 
      *
      * With Events: [CustomerEvent].

      */

     fun deleteCustomer(id: Long): Unit


         /**
      * 
      *
      */

     fun searchCustomers(input: CustomerSearchCriteria, pageable: Pageable): Page<Customer>



}
